internal/game/game.go

```diff
@@ -0,0 +1,45 @@
+package game
+
+import (
+	"github.com/gorilla/websocket"
+	"github.com/oklog/ulid/v2"
+	"go.uber.org/zap"
+)
+
+type Game struct {
+	rooms []*room
+
+	// utilities
+	logger zap.Logger
+}
+
+func New(logger zap.Logger) Game {
+	return Game{
+		rooms:  []*room{},
+		logger: logger,
+	}
+}
+
+// If a room exists, GetRoom returns a room object.
+func (g *Game) GetRoom(uid ulid.ULID) (bool, *room) {
+	for _, room := range g.rooms {
+		if room.Uid == uid {
+			return true, room
+		}
+	}
+
+	return false, nil
+}
```

Comment:
Just a thought: Since the return value is a pointer the bool return is not really necessary, I think.